src/measurement/store.ts

```diff
@@ -155,13 +181,16 @@ export class MeasurementStore {
 
 	removeDefaults (measurement: MeasurementRecord, request: MeasurementRequest): Partial<MeasurementRecord> {
 		const defaults = getDefaults(request);
+
 		// Removes `"limit": 1` from locations. E.g. [{"country": "US", "limit": 1}] => [{"country": "US"}]
-		measurement.locations = measurement.locations.map(location => location.limit === 1 ? _.omit(location, 'limit') : location);
+		if (_.isArray(measurement.locations)) {
+			measurement.locations = measurement.locations.map(location => location.limit === 1 ? _.omit(location, 'limit') : location);
+		}
```

Comment:
Actually, here, we should probably inject the original locations 🤔 Otherwise clients will need to implement special logic to handle this case and fetch the original measurement themselves. Maybe store it in the same object as the IPs?

Reply:
`redis.json.get` to the original measurement `locations` should work here.

Reply:
Yes.